cpp/modmesh/inout/plot3d.cpp

```diff
@@ -0,0 +1,150 @@
+#include <modmesh/inout/plot3d.hpp>
+
+namespace modmesh
+{
+
+namespace inout
+{
+
+Plot3d::Plot3d(const std::string & data)
+{
+    std::string line;
+    size_t total_blk_size = 0;
+    uint_type nblocks = 0;
+    const std::regex plot3d_block_delim(R"(\d+)");
+
+    stream << data;
+    // getting the first line for total blocks number of the p3d mesh file
+    std::getline(stream, line);
+    nblocks = stoul(line);
+
+    m_x_shape.remake(small_vector<size_t>{nblocks}, 0);
+    m_y_shape.remake(small_vector<size_t>{nblocks}, 0);
+    m_z_shape.remake(small_vector<size_t>{nblocks}, 0);
+    m_blk_sizes.remake(small_vector<size_t>{nblocks}, 0);
+
+    // parsing xyz dimension of each block
+    for (auto i = 0; i < nblocks; ++i)
+    {
+        std::getline(stream, line);
+        auto tokens = tokenize(line, plot3d_block_delim);
+        m_x_shape(i) = std::stoul(tokens[0]);
+        m_y_shape(i) = std::stoul(tokens[1]);
+        m_z_shape(i) = std::stoul(tokens[2]);
+        m_blk_sizes(i) = m_x_shape(i) * m_y_shape(i) * m_z_shape(i);
+        total_blk_size += m_blk_sizes(i);
+    }
+
+    m_nds.remake(small_vector<size_t>{total_blk_size, 3}, 0);
+
+    parseCoordinates(nblocks);
+    buildHexahedronElements(nblocks);
+}
+
+void Plot3d::parseCoordinates(const uint_type nblocks)
+{
+    uint_type base_idx = 0;
+    std::string line;
+    std::queue<real_type> parsing_q;
+    const std::regex plot3d_coord_delim(R"(-?\d+(\.\d+)?([eE][+-]?\d+)?|\b\d+\b)");
+
+    // TODO: The nested for-loop needs to be enhanced
```

Comment:
It would be better to be specific why the nested for loop should be enhanced.  But it's OK for now because it is short enough.